common/src/main/java/io/netty/util/ConstantPool.java

```diff
@@ -68,7 +68,7 @@ public T valueOf(String name) {
     private T getOrCreate(String name) {
         T constant = constants.get(name);
         if (constant == null) {
-            final T tempConstant = newConstant(nextId(), name);
+            final T tempConstant = newConstant(0, name);
```

Comment:
> This uniquiness was never used for id, that's why I removed it.

Perhaps not in the classes shipped with Netty however `ConstantPool` is public. What is the expected impact of this PR for application `X` which `extends ConstantPool`? How will they know that the meaning of `id` and the usage of `newConstant(int,String)` has changed?  See below for an approach which I think preserves backwards compatibility while also accommodating the cleanup in this PR.

```java
public abstract class ConstantPool<T extends Constant<T>> {
  private T getOrCreate(String name) {
  ..
  // we no longer call newConstant(id, name) in ConstantPool
  // (except for the new newConstant(name) method)
  final T tempConstant = newConstant(name);
  ..
  }

  protected T newConstant(String name) {
    return newConstant(nextId(), name);
  }
  @Deprecated
  protected abstract T newConstant(int id, String name);
}

// In ChannelOption.java

private static final ConstantPool<ChannelOption<Object>> pool = new ConstantPool<ChannelOption<Object>>() {
  @Override
  protected ChannelOption<Object> newConstant(int id, String name) {
    return new ChannelOption<Object>(id, name);
  }

  @Override
  protected ChannelOption<Object> newConstant(String name) {
    return new ChannelOption<Object>(name);
  }
};
```